runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java

```diff
@@ -90,31 +92,28 @@ public DynamicProtobufSerializer(CorfuRuntime corfuRuntime) {
         ISerializer protobufSerializer = createProtobufSerializer();
         Serializers.registerSerializer(protobufSerializer);
 
-        // Open the Registry Table.
-        CorfuTable<TableName, CorfuRecord<TableDescriptors, TableMetadata>> corfuTable = corfuRuntime.getObjectsView()
+        // Open the Protobuf Descriptor Table.
+        CorfuTable<ProtobufFileName, CorfuRecord<ProtobufFileDescriptor, TableMetadata>> descriptorTable = corfuRuntime.getObjectsView()
                 .build()
-                .setTypeToken(new TypeToken<CorfuTable<TableName, CorfuRecord<TableDescriptors, TableMetadata>>>() {
+                .setTypeToken(new TypeToken<CorfuTable<ProtobufFileName, CorfuRecord<ProtobufFileDescriptor, TableMetadata>>>() {
                 })
-                .setStreamName(getFullyQualifiedTableName(CORFU_SYSTEM_NAMESPACE, REGISTRY_TABLE_NAME))
+                .setStreamName(getFullyQualifiedTableName(CORFU_SYSTEM_NAMESPACE, PROTOBUF_DESCRIPTOR_TABLE_NAME))
                 .setSerializer(protobufSerializer)
                 .addOpenOption(ObjectOpenOption.NO_CACHE)
                 .open();
 
-        // Cache the FileDescriptorProtos from the registry table.
-        corfuTable.forEach((tableName, value) -> {
-            TableDescriptors tableDescriptors = value.getPayload();
-            tableDescriptors.getFileDescriptorsMap().forEach((fdName, fileDescriptorProto) -> {
-                String protoFileName = fileDescriptorProto.getName();
-                // Looks like protobuf file descriptors are randomly truncating the path.
-                // This causes dynamicProtobufSerializer to fail since the full path is needed.
-                if (protoFileName.equals("corfu_options.proto")) {
-                    fdProtoMap.putIfAbsent(protoFileName, fileDescriptorProto);
-                    // Until the truncating issue can be addressed, manually add both paths.
-                    protoFileName = "corfudb/runtime/corfu_options.proto";
-                }
-                fdProtoMap.putIfAbsent(protoFileName, fileDescriptorProto);
-                identifyMessageTypesinFileDescriptorProto(fileDescriptorProto);
-            });
+        // Cache the FileDescriptorProtos from the protobuf descriptor table.
+        descriptorTable.forEach((fdName, fileDescriptorProto) -> {
+            String protoFileName = fileDescriptorProto.getPayload().getFileDescriptor().getName();
+            // Looks like protobuf file descriptors are randomly truncating the path.
+            // This causes dynamicProtobufSerializer to fail since the full path is needed.
+            if (protoFileName.equals("corfu_options.proto")) {
+                fdProtoMap.putIfAbsent(protoFileName, fileDescriptorProto.getPayload().getFileDescriptor());
+                // Until the truncating issue can be addressed, manually add both paths.
+                protoFileName = "corfudb/runtime/corfu_options.proto";
```

Comment:
What does this mean, what is the truncating issue?

Reply:
since we are within the corfu runtime, the Any module produces the corfu_options without the path prefix (truncation), but later in another module we require the entire prefix to decode causing deserialization errors, hence this fix-up for corfu_options.proto

Reply:
Please capture this in comments.

Reply:
done